astlib/history.ml

```diff
@@ -97,21 +97,26 @@ let create ~versioned_grammars ~conversions =
 let versioned_grammars t =
   Array.to_list (Array.map2 t.versions t.grammars ~f:(fun v g -> v, g))
 
+let apply_conversion conversion node ~to_node ~of_node =
+  conversion.f node
+    ~of_src_node:(of_node ~version:conversion.src_version)
```

Comment:
Isn't it the other way around? `of_node` is used to "build" new nodes and therefore should target `dst` while `to_node` is used to inspect nodes which we have to assume are versioned as `src`, so that we know how they are formed.

Reply:
> `of_node` is used to "build" new nodes

Sorry I should have said to build new `'a`s

Reply:
`'a` being the version wrapped node

Reply:
Hmm, sounds like we should clarify some of these names while we're at it.

Reply:
And it's clearly bad that I broke it and no tests complained. In fact, I should probably make some tests first that demonstrate correct behavior, and THEN do this fix.